ion/binaryreader_test.go

```diff
@@ -16,13 +16,171 @@
 package ion
 
 import (
+	"bytes"
 	"fmt"
 	"math"
 	"math/big"
 	"testing"
 	"time"
 )
 
+const binaryFilePath = "testdata/binary/"
+
+func TestReadBinarySingleSymbol(t *testing.T) {
+	fileBytes := loadFile(t, binaryFilePath+"single_symbol.bindat")
+	r := NewReader(bytes.NewReader(fileBytes))
+	assertSingleSymbolTest(t, r)
+}
+
+func TestReadBinaryFieldNameSymbolAndSymbolValues(t *testing.T) {
+	table1 := NewSharedSymbolTable("table1", 1, []string{"s1", "s2"})
+	table2 := NewSharedSymbolTable("table2", 1, []string{"s3", "s4"})
+	catalog := NewCatalog(table1, table2)
+
+	buf := bytes.Buffer{}
+	bw := NewBinaryWriter(&buf, table1, table2)
+	err := bw.BeginStruct()
+	if err != nil {
+		t.Fatal("unable to write struct")
+	}
+	err = bw.FieldName("s1")
+	if err != nil {
+		t.Fatal("unable to write field name")
+	}
+	err = bw.WriteSymbol("s2")
+	if err != nil {
+		t.Fatal("unable to write string")
+	}
+	err = bw.FieldName("s3")
+	if err != nil {
+		t.Fatal("unable to write field name")
+	}
+	err = bw.WriteSymbol("s4")
+	if err != nil {
+		t.Fatal("unable to write string")
+	}
+	err = bw.EndStruct()
+	if err != nil {
+		t.Fatal("unable to end struct")
+	}
+	err = bw.Finish()
+	if err != nil {
+		t.Fatal("unable to create finish")
+	}
+
+	r := NewReaderBytes(buf.Bytes())
+	r.Next()
+	err = r.StepIn()
+	if err != nil {
+		t.Fatal("unable to step in")
+	}
+
+	r.Next()
+	currentType := r.Type()
+	if currentType != SymbolType {
+		t.Fatal("current type is not symbol.")
+	}
+	fns, err := r.FieldNameSymbol()
+	if err != nil {
+		t.Fatal("unable to get field name symbol")
+	}
+	if fns.Text != nil {
+		t.Fatal("field name symbol text is incorrect")
+	}
+	if fns.LocalSID != 10 {
+		t.Fatal("field name symbol sid is incorrect")
+	}
```

Comment:
Creating versions of `_symbolAF` and friends that deal in SymbolTokens might make the intent of these sections more clear.

Reply:
Modified _symbolAF to support SymbolToken validation. Also added _nextF to support FieldNameSymbol validation